ctrl-d on `ipfs block stat` causes a panic

Version/Platform/Processor information (from `ipfs version --all`):

```
» ipfs version --all
go-ipfs version: 0.4.3-dev-d742fb1
Repo version: 4
System version: amd64/darwin
Golang version: go1.6.2
```

Type (bug, feature, meta, test failure, question): bug
Area (api, commands, daemon, fuse, etc): cli
Priority (from P0: functioning, to P4: operations on fire): P1

Description:

```
» ipfs block stat
ipfs: Reading from /dev/stdin; send Ctrl-d to stop.
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0xa7b560, 0xc82000e090)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/ipfs/go-ipfs/core/commands.glob.func12(0x2149848, 0xc82015e600, 0x2149ae0, 0xc82025d810)
        /Users/david/code/go-projects/src/github.com/ipfs/go-ipfs/core/commands/block.go:61 +0x224
github.com/ipfs/go-ipfs/commands.(*Command).Call(0x125ffe0, 0x2149848, 0xc82015e600, 0x0, 0x0)
        /Users/david/code/go-projects/src/github.com/ipfs/go-ipfs/commands/command.go:111 +0x472
main.callCommand(0x2149a50, 0xc82032a0c0, 0x2149848, 0xc82015e600, 0x125ffe0, 0x123d7c0, 0x0, 0x0, 0x0, 0x0)
        /Users/david/code/go-projects/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:346 +0xa84
main.(*cmdInvocation).Run(0xc82032a000, 0x2149a50, 0xc82032a0c0, 0x0, 0x0, 0x0, 0x0)
        /Users/david/code/go-projects/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:189 +0x1e7
main.main()
        /Users/david/code/go-projects/src/github.com/ipfs/go-ipfs/cmd/ipfs/main.go:154 +0x660
```

--- a/commands/request.go
+++ b/commands/request.go
@@ -240,13 +240,19 @@ func (r *request) VarArgs(f func(string) error) error {
 		return err
 	}
 
+	var any bool
 	scan := bufio.NewScanner(fi)
 	for scan.Scan() {
+		any = true
 		err := f(scan.Text())
 		if err != nil {
 			return err
 		}
 	}
+	if !any {
+		return f("")
+	}
+
 	return nil
 }
 

--- a/test/sharness/t0050-block.sh
+++ b/test/sharness/t0050-block.sh
@@ -39,4 +39,12 @@ test_expect_success "'ipfs block get' output looks good" '
   test_cmp expected_stat actual_stat
 '
 
+test_expect_success "'ipfs block stat' with nothing from stdin doesnt crash" '
+	test_expect_code 1 ipfs block stat < /dev/null 2> stat_out
+'
+
+test_expect_success "no panic in output" '
+	test_expect_code 1 grep "panic" stat_out
+'
+
 test_done

